samples/openapi3/client/petstore/python-experimental/tests/test_deserialization.py

```diff
@@ -161,3 +161,26 @@ def test_deserialize_mammal(self):
         self.assertTrue(isinstance(deserialized, petstore_api.Zebra))
         self.assertEqual(deserialized.type, zebra_type)
         self.assertEqual(deserialized.class_name, class_name)
+
+    def test_deserialize_oneof_reference(self):
+
+        inst = petstore_api.Drawing(
+            # 'main_shape' has type 'Shape', which is a oneOf [triangle, quadrilateral]
+            # composed schema. So we should be able to assign a petstore_api.Triangle
+            # to a 'main_shape'.
+            main_shape=petstore_api.Triangle(
+                shape_type="Triangle",
+                triangle_type="IsoscelesTriangle"
+            ),
+            shapes=[
+                petstore_api.Triangle(
+                    shape_type="Triangle",
+                    triangle_type="IsoscelesTriangle"
+                ),
+                petstore_api.Shape(
+                    shape_type="Quadrilateral",
+                    quadrilateral_type="ComplexQuadrilateral"
+                ),
+            ],
+        )
+        assert isinstance(inst, petstore_api.BiologyHominid)
```

Comment:
This line is hard for me to understand. Why are we expecting the instance type to be BiologyHominid. Should it be Drawing?

Also, because Drawing is an object type schema (ModelNormal subclass) in python, and doesn't have a discriminator, we will never pick a different class when moving through the new method, so we are sure that instances that we make from it will always be of type Drawing.

Reply:
> This line is hard for me to understand. Why are we expecting the instance type to be BiologyHominid. Should it be petstore_api .Drawing?

This is a typo.

> 
> Also, because Drawing is an object type schema (ModelNormal subclass) in python, and doesn't have a discriminator, we will never pick a different class when moving through the new method, so we are sure that instances that we make from it will always be of type Drawing.

Sure, it's just that I was planning to add more assertions, and it does not hurt to add an assertion.

Reply:
Nope it doesn't hurt